api/src/wfl/module/xx.clj

```diff
@@ -2,196 +2,167 @@
-                        (keep input?)
-                        (into {}))
-            more   (->> suffix
-                        keys
-                        (remove done)
-                        (take max)
-                        (map (fn [base] (str base (suffix base)))))]
-        (run! (partial hold-workflow environment bucket out-gs) more)))))
-
-(defn release-some-on-hold-workflows
-  "Submit up to MAX-STRING workflows from 'On Hold' in ENVIRONMENT."
-  [environment max-string]
-  (letfn [(release! [id] (prn (cromwell/release-hold environment id)))]
-    (let [max (util/is-non-negative! max-string)]
-      (->> {:pagesize max :status "On Hold" :label cromwell-label}
-           (cromwell/query environment)
-           (map :id)
-           (take max)
-           (run! release!)))))
-
-(defn run
-  "Reprocess the BAM or CRAM files described by ARGS."
-  [& args]
-  (try
-    (let [env (wfl/throw-or-environment-keyword! (first args))]
-      (apply (case (count args)
-               4 on-hold-some-workflows
-               3 (partial all/report-status cromwell-label)
-               2 release-some-on-hold-workflows
-               (throw (IllegalArgumentException.
-                       "Must specify 2 to 4 arguments.")))
-             env (rest args)))
-    (catch Exception x
-      (log/error description)
-      (throw x))))
+(def ^:private scatter-settings-defaults
+  {:haplotype_scatter_count     50
+   :break_bands_at_multiples_of 0})
+
+(def ^:private papi-settings-defaults
+  {:agg_preemptible_tries 3
+   :preemptible_tries     3})
+
+(def ^:private workflow-defaults
+  (let [hg38 "gs://gcp-public-data--broad-references/hg38/v0/"]
+    {:unmapped_bam_suffix  ".unmapped.bam"
+     :cram_ref_fasta       (str hg38 "Homo_sapiens_assembly38.fasta")
+     :cram_ref_fasta_index (str hg38 "Homo_sapiens_assembly38.fasta.fai")
+     :bait_set_name        "whole_exome_illumina_coding_v1"
+     :bait_interval_list   (str hg38 "HybSelOligos/whole_exome_illumina_coding_v1/whole_exome_illumina_coding_v1.Homo_sapiens_assembly38.baits.interval_list")
+     :target_interval_list (str hg38 "HybSelOligos/whole_exome_illumina_coding_v1/whole_exome_illumina_coding_v1.Homo_sapiens_assembly38.targets.interval_list")
+     :references           references-defaults
+     :scatter_settings     scatter-settings-defaults
+     :papi_settings        papi-settings-defaults}))
+
+(defn- get-cromwell-environment [{:keys [cromwell]}]
+  (let [envs (all/cromwell-environments #{:xx-dev :xx-prod} cromwell)]
+    (if-not (empty? envs) ;; assuming prod and dev urls are different
+      (first envs)
+      (throw
+        (ex-info "No environment matching Cromwell URL."
+          {:cromwell cromwell})))))
+
+(defn- make-cromwell-inputs [environment inputs]
+  (->
+    (env/stuff environment)
+    (select-keys [:google_account_vault_path :vault_token_path])
+    (merge inputs)
+    (util/prefix-keys (keyword pipeline))))
+
+;; visible for testing
```

Comment:
I appreciate the docstring here. Thanks.
Maybe `make-list-of-samples` or `input-items->list-of-samples` or sumpin?